src/SeleniumLibrary/keywords/cookie.py

```diff
@@ -55,24 +55,55 @@ def get_cookie_value(self, name):
 
     @keyword
     def get_cookie(self, name):
-        """Returns the cookie found with `name` in extended variable format.
-
-        If no cookie is found with `name`, this keyword fails.
+        """Returns a cookie object found with ``name``.
+
+        If no cookie is found with ``name``, keyword fails. The cookie object
+        contains details about the cookie. Attributes available in the object
+        are documented in the table below.
+        | = Attribute = |             = Explanation =                            |
+        | domain        | Specifies hosts to which the cookie will be sent       |
+        | expiry        | The maximum lifetime of the cookie as EPOCH            |
+        | httpOnly      | HttpOnly cookie cannot be accessed by client-side APIs |
+        | name          | The name of a cookie                                   |
+        | path          | Indicates a URL path, example /                        |
+        | secure        | Cookie will be send only by using secure connection    |
+        | value         | Value of the cookie                                    |
+        | full_info     | All the above attributes joined as string              |
+
+        New in SeleniumLibrary 3.0
+
+        Example:
+        | Add Cookie      | foo             | bar |
+        | ${cookie} =     | Get Cookie      | foo |
+        | Should Be Equal | ${cookie.value} | foo |
+        | Should Be Equal | ${cookie.name}  | bar |
         """
         cookie = self.browser.get_cookie(name)
         if cookie:
-            cookie_information = CookieInformation(cookie)
-            return cookie_information
+            return CookieInformation(
+                cookie.get('domain', None), cookie.get('expiry', None),
+                cookie.get('httpOnly', None), cookie['name'],
+                cookie.get('path', None), cookie.get('secure', None),
+                cookie['value'])
         raise ValueError("Cookie with name %s not found." % name)
 
     @keyword
     def add_cookie(self, name, value, path=None, domain=None, secure=None,
                    expiry=None):
         """Adds a cookie to your current session.
 
-        "name" and "value" are required, "path", "domain", "secure" and
-         "expiry" are optional.  Expiry supports the same formats as
-         the DateTime library.
+        ``name`` and ``value`` are required, ``path``, ``domain``, ``secure``
+        and ``expiry`` are optional.  Expiry supports the same formats as
+        the [http://robotframework.org/robotframework/latest/libraries/DateTime.html|DateTime]
+        library and is converted to EPOCH timestamp which is supported by the
+        Selenium.
```

Comment:
The note how the value is internally turned into an EPOCH timestamp isn't too relevant.